### PR TITLE
added new inlet outlet files

### PR DESCRIPTION
**Pull request by psnbaba**

### CODE REVIEW

File: pysph/examples/flow_past_cylinder_2d.py [hunk 1]
```diff
@@ -15,43 +8,56 @@
...
 umax = 1.0
 c0 = 10 * umax
 p0 = rho * c0 * c0
 
 
-class ExtrapolateUhat(Equation):
-    def initialize(self, d_idx, d_uhat, d_wij):
-        d_uhat[d_idx] = 0.0
-        d_wij[d_idx] = 0.0
+class SolidWallNoSlipBCReverse(Equation):
```
Comment (prabhuramachandran) [inline pysph/examples/flow_past_cylinder_2d.py]: Is this the correct way to find the forces?  Does this need to be updated from the other repos?

File: pysph/examples/trivial_inlet_outlet.py [hunk 2]
```diff
@@ -121,8 +136,8 @@ def create_solver(self):
         self.iom = self._create_inlet_outlet_manager()
         kernel = CubicSpline(dim=2)
         integrator = PECIntegrator(
-            fluid=InletOutletStepEDAC(), inlet=InletOutletStepEDAC(),
-            outlet=InletOutletStepEDAC()
+            fluid=InletStep(), inlet=InletStep(),
+            outlet=InletStep()
```
Comment (prabhuramachandran) [inline pysph/examples/trivial_inlet_outlet.py]: Should this be `OutletStep`?

Comment (prabhuramachandran) [reply pysph/examples/trivial_inlet_outlet.py]: If not the imports need to be fixed.

File: pysph/sph/bc/mirror/simple_inlet_outlet.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+                ))
+            g00.append(CopyNormalsandDistances(
+                dest=all_pairs[info.pa_name], sources=[info.pa_name]))
+            i = i + 1
+
+        equations.append(Group(equations=g00, real=False))
+
+        g02 = []
+        for name in all_ghosts:
+            g02.append(UpdateD(dest=name, sources=self.fluids, dim=self.dim))
```
Comment (prabhuramachandran) [inline pysph/sph/bc/mirror/simple_inlet_outlet.py]: The name `UpdateD` is not very clear or descriptive.  Perhaps a better name is needed?

File: pysph/examples/trivial_inlet_outlet.py (hunk 2 above)
Comment (psnbaba) [reply pysph/examples/trivial_inlet_outlet.py]: yes it should be OutletStep.

File: pysph/sph/bc/mirror/simple_inlet_outlet.py (hunk 3 above)
Comment (psnbaba) [reply pysph/sph/bc/mirror/simple_inlet_outlet.py]: Changed to UpdateMomentMatrix

File: pysph/examples/flow_past_cylinder_2d.py (hunk 1 above)
Comment (psnbaba) [reply pysph/examples/flow_past_cylinder_2d.py]: Yes, It is the same method used in inlet outlet repository.

File: docs/source/tutorial/circular_patch_simple.rst [hunk 4]
```diff
@@ -482,6 +482,23 @@ the interpolation is desired.  This can also be done with the constructor as::
 
     interp = Interpolator(list(parrays.values()), x=x, y=y, z=z)
 
+There are some cases, where one may require a higher order interpolation or
+gradient approximation of the property. This can be done by passing a
+``method`` for interpolation to the interplator as::
+
+    interp = Interpolator(list(parrays.values()), num_points=10000, method='order1')
+
+Currently, PySPH have three method of interpolation namely ``shepard``,
```
Comment (prabhuramachandran) [inline docs/source/tutorial/circular_patch_simple.rst]: have three method of ... -> has three methods of ...

Comment (psnbaba) [reply docs/source/tutorial/circular_patch_simple.rst]: done

**Pull request merged**